### DISCUSSION TITLE
Anyone use Sunshine/Moonlight for pair programming?

### DISCUSSION THREAD

**Discussion by unphased** [LizardByte/community#278] [Sunshine]
As a twitchy gamer and also a software engineer, I absolutely prefer to have the best technology available to use for productivity purposes. 

I have tried out Tuple in the past (colleague and I both on macOS) and it worked quite well and with better latency and quality than anything else we tested at the time (Google Meet, Zoom, Slack, AnyDesk...). I also know that Tuple recently finally provides support for Linux. 

However I'm still convinced that the quality and latency of Sunshine/Moonlight is going to be vastly superior still compared to something like Tuple because the primary use case (games) requires pushing the performance of the stream to the maximum possible, and that the only possible way to make a connection is a direct P2P connection cutting out any possible middleman that would introduce latency. 

No productivity focused tool like this will provide 120hz+ refresh rate stream. There's no market for that feature in that space. But I know that with Sunshine if I want to collaborate with a colleague or friend on something, if our network connection can allow for it, nothing stops us from having a 120hz or higher shared remote stream. Sunshine/Moonlight is firmly in another league compared to Tuple, and Tuple is already in another league compared to the other competing solutions.

Since a product such as Tuple is geared toward collaboration and productivity, it obviously (at least currently) has more of a featureset that is desirable and missing from Sunshine in this use case. But I definitely feel that there is a lot of value that exists in Sunshine/Moonlight that is still locked away when it comes to productivity. What do you think? I also really like how Moonlight has such great platform support, so in this area it is already so far ahead of other solutions. The only one that has near parity is AnyDesk. And to be honest I have been using AnyDesk a lot to do different kinds of work as well as access all of my machines, but I am begining my transition to Sunshine/Moonlight for this use case, to take advantage of the amazing quality. Needing to fork Moonlight to hack in more client features won't hold me back from going down this road!

How receptive would the project's (this one and Moonlight's) maintainers be to PR's that flesh out authentication and access control and maybe little features like handing off control in order to be able to display the position of the passive user's cursor location? (I won't even get into adding painting ability. That's more or less a gimmick. Fun but not valuable. Also I'll definitely say that audio session and chat are both out of scope.) So far from my testing, a lot of the bare necessities already work pretty great. If the passive user doesn't insist on wiggling their mouse, it's quite usable already. The only thing that's cumbersome is that for it to be actually secure, we have to micromanage the authentications. So I would be targeting that first for sure.

**Comment by ReenigneArcher** [comment]
Sunshine is definitely optimized for gaming, not productivity features.

And regarding changing protocols, I'd suggest joining our discord server to discuss. It will be easier to get feedback from the main devs there.

**Reply by unphased** [reply]
Thanks for the suggestion to join the discord. I will do that. So i was looking into tailscale / openziti for this but soon it became obvious that security is going to have to be intimately connected to the transmission protocols as layering the stream on top of a non optimized transport layer for security is going to leave heaps of performance on the table. so the path is a bit murkier than i thought it was gonna be at first...

[a comment by an automated account is not shown]
